File: foody-master/Back End/src/main/java/com/food/service/OrderService.java
package com.food.service;

import java.util.List;

import com.food.model.Order;

public interface OrderService {
	public Order addOrder(Order order);
	public List<Order> getAllOrder();
	public Order updateOrder(Order order);
	public void removeOrder(int id);
	public List<Order> getOrderById(int id);
}

File: foody-master/Front End/src/app/services/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { orderUrl } from '../config/api'
import { Order } from '../model/order';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private http: HttpClient) { }

  addOrder(order: Order){
    return this.http.post(orderUrl,order);
  }

  updateOrder(order: Order){
    return this.http.put(orderUrl,order);
  }

  getAllOrder():Observable<Order[]>{
    return this.http.get<Order[]>(orderUrl);
  }

  getOrderById(id: number):Observable<Order[]>{
    return this.http.get<Order[]>(orderUrl+"/"+id);
  }
}
